### PR TITLE
Add pagination tests + fix pagination

### PR DESCRIPTION
**Pull request by bidoubiwa**

### CODE REVIEW

File: src/transformers/__tests__/utils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { MeiliSearch } from 'meilisearch'
+const HOST = 'http://localhost:7700'
+// const apiKey = ''
```
Comment (curquiza) [inline src/transformers/__tests__/utils.ts:3]:
```suggestion
```

File: src/transformers/__tests__/pagination.tests.ts [hunk 2]
```diff
@@ -1,5 +1,172 @@
...
+    numberPages: 0,
+  },
+  // Not an Algolia behavior. Algolia return an error:
+  // "Value too small for \"hitsPerPage\" parameter, expected integer between 0 and 9223372036854775807",
+  {
+    hitsPerPage: -1,
+    hitsLength: 20,
+    numberPages: 0,
+  },
+  // Not an Algolia behavior. Algolia return an error:
```
Comment (curquiza) [inline src/transformers/__tests__/pagination.tests.ts:37]:
```suggestion
  // Not an Algolia behavior. Algolia returns an error:
```

File: src/transformers/__tests__/pagination.tests.ts [hunk 3]
```diff
@@ -1,5 +1,172 @@
...
+    hitsPerPage: 20,
+    hitsLength: 0,
+    numberPages: 0,
+  },
+  {
+    hitsPerPage: 0,
+    hitsLength: 0,
+    numberPages: 0,
+  },
+  // Not an Algolia behavior. Algolia return an error:
```
Comment (curquiza) [inline src/transformers/__tests__/pagination.tests.ts:30]:
```suggestion
  // Not an Algolia behavior. Algolia returns an error:
```

File: src/transformers/__tests__/pagination.tests.ts [hunk 4]
```diff
@@ -1,5 +1,172 @@
...
+    returnedHits: [],
+  },
+  {
+    hits: [{ id: 1 }, { id: 2 }, { id: 3 }],
+    page: 2,
+    hitsPerPage: 0,
+    returnedHits: [],
+  },
+  // Wrong types
+  // Not an Algolia behavior. Algolia return an error:
```
Comment (curquiza) [inline src/transformers/__tests__/pagination.tests.ts:133]:
```suggestion
  // Not an Algolia behavior. Algolia returns an error:
```

File: src/transformers/__tests__/utils.ts (hunk 1 above)
Comment (curquiza) [inline src/transformers/__tests__/utils.ts:3]:
```suggestion

```

**Review by curquiza — COMMENTED** [review]
Nice display and the list of tests seems exhaustive :)

Why do you name the variables `algolia...` in your tests, like `algoliaNumberPagesBehavior`? I understand this is a list for our tests with the parameters and the expected return. No need to mention algolia, right?

**Comment by bidoubiwa** [comment]
> Why do you name the variables algolia... in your tests, like algoliaNumberPagesBehavior? I understand this is a list for our tests with the parameters and the expected return. No need to mention algolia, right?

Okey! Since it was the behavior of Algolia I thought why not name it like that but I understand your point.

**Review by curquiza — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**